frontline-functions-quickstart/functions/callbacks/twilio-conversations.protected.js

```diff
@@ -0,0 +1,126 @@
+       *
+       * It is required especially if Frontline Inbound Routing is enabled
+       * so that when the worker will be added to the conversation, they will
+       * see the friendly_name and avatar of the conversation.
+       *
+       * More info about the `onConversationAdd` webhook: https://www.twilio.com/docs/conversations/conversations-webhooks#onconversationadd
+       * More info about handling incoming conversations: https://www.twilio.com/docs/frontline/handle-incoming-conversations
+       */
+      const customerNumber = event['MessagingBinding.Address'];
+      const isIncomingConversation = Boolean(customerNumber);
+
+      if (isIncomingConversation) {
+        try {
+          const customerDetails =
+            (await getCustomerByNumber(context, customerNumber)) || {};
+
+          const conversationProperties = {
+            friendly_name: customerDetails.display_name || customerNumber,
+            attributes: JSON.stringify({
+              avatar: customerDetails.avatar,
+            }),
+          };
+
+          return callback(null, conversationProperties);
+        } catch (err) {
+          return callback(err);
+        }
+      }
+      break;
+    }
+    case 'onParticipantAdded': {
+      console.log('[[[onParticipantAdded]]]');
+      /**
+       * POST-WEBHOOK
+       *
+       * This webhook will be called when a participant added to a conversation
+       * including customer in which we are interested in.
+       *
+       * It is required to add customer_id information to participant and
+       * optionally his display_name and avatar.
+       *
+       * More info about the `onParticipantAdded` webhook: https://www.twilio.com/docs/conversations/conversations-webhooks#onparticipantadded
+       * More info about the customer_id: https://www.twilio.com/docs/frontline/my-customers#customer-id
+       * And more here you can see all the properties of a participant which you can set: https://www.twilio.com/docs/frontline/data-transfer-objects#participant
+       */
+      const conversationSid = event.ConversationSid;
+      const participantSid = event.ParticipantSid;
+      const customerNumber = event['MessagingBinding.Address'];
+      const isCustomer = customerNumber && !event.Identity;
+
+      if (isCustomer) {
+        try {
+          const customerParticipant = await client.conversations
+            .conversations(conversationSid)
+            .participants.get(participantSid)
+            .fetch();
+
+          const customerDetails =
+            (await getCustomerByNumber(context, customerNumber)) || {};
+          await setCustomerParticipantProperties(
+            customerParticipant,
+            customerDetails
+          );
+          return callback(null, 'success');
+        } catch (err) {
+          return callback(err);
+        }
+      }
+      return callback(404, `Something went wront`);
+    }
+
+    default: {
+      return callback(422, `Unknown location: location`);
```

Comment:
maybe something like `unhandled webhook event`. Location is a term used in frontline callbacks, this is conversations webhooks.